lib/datadog/core/configuration/settings.rb

```diff
@@ -634,7 +634,38 @@ def initialize(*_)
           #
           # @default `nil`.
           # @return [String,nil]
-          option :service
+          option :service do |o|
+            o.type :string, nilable: true
+          end
+
+          # Declare additional service names to bind to remote configuration. Use when
+          # integrations are configured to use names that differ from DD_SERVICE.
+          #
+          # @default `nil`.
+          # @return [Array<String>,nil]
```

Comment:
If we would like to make this option not public, let's add the `# @!visibility private` tag to it.

Reply:
Good point, although I'm not sure making it non-public is a good thing either, so I tried to describe it in a more user friendly way.

Reply:
What does this mean for an option like this one to be "non-public"?

Reply:
The description is clear enough of what the code would do.

Reply:
> What does this mean for an option like this one to be "non-public"?

By making it not public, we can change the name/implementation without worrying about breaking the customer's application.

We could name it something like `experimental_extra_services` to let customers know that this option is somewhat not fully clear. Still we will need help to rename it easily in the future. 

What about adding as part of the comment that the use of this option is only meant for edge cases in which RC is not working due to lack of understanding between the tracer and the backend, and that we might remove it in the future once is no longer necessary?

Reply:
> What about adding as part of the comment that the use of this option is only meant for edge cases in which RC is not working due to lack of understanding between the tracer and the backend, and that we might remove it in the future once is no longer necessary?

This looks a bit weird -- seems like a "public-ish/private-ish"? My question was exactly that it's weird to add a version and for it to not be public.

We may still recommend strongly that people don't use it (via docs, logs, etc), but if it's in the settings, it seems to me that it would automatically be public?

Reply:
I agree with Ivo, e.g having support direct users to use a non-public API sounds odd; I can see users feeling less confident with that than if it is public.